File: rust/src/5/2525.rs
use std::io::stdin;

fn main() {
    // get numbers
    let (mut hour, mut minute) = {
        let mut input = String::new();
        stdin().read_line(&mut input).unwrap();
        let numbers = input
            .split_whitespace()
            .map(|x| x.parse::<i32>().unwrap())
            .collect::<Vec<i32>>();
        (numbers[0], numbers[1])
    };

    let minutes = {
        let mut input = String::new();
        stdin().read_line(&mut input).unwrap();
        input.trim().parse::<i32>().unwrap()
    };

    // calculate
    minute += minutes;

    // carry
    while minute >= 60 {
        minute -= 60;
        hour += 1;
    }

    while hour >= 24 {
        hour -= 24;
    }

    // print
    println!("{} {}", hour, minute);
}

File: c/5/2525.c
#include <stdio.h>

int main() {
    int hour, minute;
    int delta;
    scanf("%d %d", &hour, &minute);
    scanf("%d", &delta);

    minute += delta;

    hour += minute / 60;
    minute %= 60;
    hour %= 24;

    printf("%d %d\n", hour, minute);

    return 0;
}
